cdap-ui/app/cdap/components/SecureKeys/SecureKeyDetails/index.tsx

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import * as React from 'react';
+
+import withStyles, { StyleRules, WithStyles } from '@material-ui/core/styles/withStyles';
+
+import ArrowBackIcon from '@material-ui/icons/ArrowBack';
+import Button from '@material-ui/core/Button';
+import Divider from '@material-ui/core/Divider';
+import FileCopyOutlinedIcon from '@material-ui/icons/FileCopyOutlined';
+import FormControl from '@material-ui/core/FormControl';
+import IconButton from '@material-ui/core/IconButton';
+import InputAdornment from '@material-ui/core/InputAdornment';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+import classnames from 'classnames';
+import { copyToClipBoard } from 'services/Clipboard';
+
+const styles = (theme): StyleRules => {
+  return {
+    secureKeysTitle: {
+      paddingTop: theme.spacing(1),
+    },
+    divider: {
+      width: '100vw',
+    },
+    secureKeyDetails: {
+      margin: `${theme.Spacing(3)}px 0px`,
+      border: `1px solid ${theme.palette.grey[300]}`,
+      borderRadius: '6px',
+      padding: theme.spacing(3),
+      display: 'grid',
+      gridTemplateColumns: 'repeat(4, 1fr)',
+    },
+    keyID: {
+      gridColumnStart: '1',
+      gridColumnEnd: '2',
+    },
+    details: {
+      gridColumnStart: '2',
+      gridColumnEnd: '5',
+    },
+    secureKeyActionButton: {
+      margin: theme.spacing(1),
+      float: 'right',
+    },
+    secureKeyInput: {
+      margin: `${theme.Spacing(1)}px 0px`,
+      whiteSpace: 'nowrap',
+      wordWrap: 'normal',
+      paddingRight: '12px',
+      letterSpacing: '.00625em',
+      fontSize: '1rem',
+      lineHeight: '1.5rem',
+    },
+  };
+};
+
+interface ISecureKeysDetailsProps extends WithStyles<typeof styles> {
+  state: any;
+  onBackButtonClick: () => void;
+  openDeleteDialog: (index: number) => void;
+  openEditDialog: (index: number) => void;
+}
+
+const SecureKeysDetailsView: React.FC<ISecureKeysDetailsProps> = ({
```

Comment:
I don't think you need a details view for secure keys. Can the detail view and the edit mode be combined? There's not a lot of information needed to be shown

Reply:
Will discuss with @leacd on this